client/coral-plugin-commentbox/CommentBox.js

```diff
@@ -68,6 +101,15 @@ class CommentBox extends Component {
     this.setState({body: ''});
   }
 
+  addCommentHooks = (hooks = {}) => {
+    if (typeof hooks === 'object') {
```

Comment:
I imagine sth like this:
```js
this.hookId = registerHook(type, callback);
unregisterHook(this.hookId);
```
Also I suggest an unregister at `componentWillUnmount`